alf/utils/distributed.py

```diff
@@ -59,6 +59,16 @@ def __init__(self, module: torch.nn.Module, perform: Callable[..., Any]):
             if type(value) is not torch.Tensor:
                 self._ddp_params_and_buffers_to_ignore.append(name)
 
+        # We also need to ignore all the buffers that is under the replay buffer
+        # of the module (e.g. when the module is an Algorithm) for DDP, because
+        # we do not want DDP to synchronize replay buffers across processes.
+        # Those buffers are not registered in the state_dict() because of Alf's
+        # special treatment but can be found under named_buffers(). We do not
+        # want DDP to synchronize replay buffers.
+        for name, _ in self.named_buffers():
```

Comment:
This seems a little unreliable to me. How about something like the following:
```python
for module in self.modules():
    if isinstance(module, ReplayBuffer):
       ignore buffers in module
```

Reply:
Updated the PR so that it hunts for `ReplayBuffer` module and ignores the names of all the buffers that is under the `ReplayBuffer` module. Tested that the undesired buffers are correctly removed.